source/schemas/types.txt

```diff
@@ -509,3 +509,80 @@ with the ``title`` field set to the embedded object's type name:
             }
       }
    }
+
+.. _geospatial:
+
+Geospatial Data
+---------------
+
+Geospatial data describes points and other data on the earth's surface. App 
+Services supports defining GeoJSON Point data in a schema. 
+
+GeoJSON Point
+~~~~~~~~~~~~~
+
+A GeoJSON Point is a single location on the Earth's surface, described by its 
+longitude and latitude. To ensure wide compatibility, App Services stores these 
+points in accordance to the `GeoJSON spec <https://datatracker.ietf.org/doc/html/rfc7946>`_.
+To store a GeoPoint in Atlas, you must follow these rules:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- It must contain the following fields:
+
+  - ``type``, which is always set to "Point"
+  - ``coordinates``, of type ``array``. The ``coordinates`` array contains 
+    exactly 2 double values. The first double is the longitude of the point and 
+    the second is the latitude.
+
+- Additional properties are allowed.
+ 
+The following is the schema of an object where the "location" property is 
+a GeoPoint:
+
+.. code-block:: json
+
+   {
+     "title": "MyObject",
+     "properties": {
+       "location": {
+         "bsonType": "object",
+         "required": [ "type" ],
+         "properties": {
+           "type": {
+             "bsonType": "string",
+             "const": "Point"
```

Comment:
[nit] The "const": "Point" is not necessary in order for the server to interpret an embedded object as a GeoJSON Point. I think we should remove it

Reply:
@jonatlee I'm confused; isn't the "type" property required _and_ also must be set to "point"? The spec suggests that's the case.

Reply:
[This](https://docs.google.com/document/d/1mOV04pa_Wl4uC5l6wBooQWyv2OVsBll6gtKaWl-ugGE/edit#bookmark=id.y8wrmys9ft74) is the expected minimum JSON schema structure for defining a GeoPoint property. We originally were going to include "enum": ["Point"] similar to the above but we decided that it wasn't necessary. The server performs validation on top of the JSON schema validation that ensures that the "type" field will only ever get set to "Point". That is what 3.) in this comment https://github.com/mongodb/docs-app-services/pull/597#discussion_r1347878102

Reply:
@jonatlee one last question on the schema: "type" is the only required property? Not "coordinates", too?

Reply:
Arrays must be optional in order to be sync compatible, so "coordinates" should not be specified as part of the required fields lists. The server will ensure that writes to "coordinates" will always leave it in a valid state though (array of doubles with length >= 2, longitude is in the range [-180, 180] and latitude is in the range [-90, 90])